### PR TITLE
Include logic to send solid colors as foreground to shine and add sample profiles

### PR DESCRIPTION
**Pull request by thomazmoura**
Include the logic to send a solid color from qmk to shine. 
Include 2 new profiles based on default to show use cases with the new logic for newcomers and my personal profile as a sample of a more advanced scenario.

## Description

Include the logic to send a solid color from qmk to shine. That solid color will act as a foreground (will override the current profile) until reset (witch will reactivate the current profile).
This functionality depends on changes made for shine as well.

Include 3 new profiles:

* default-full-caps -> same as default, but with the logic of using the red foreground color on caps lock.
* default-layer-indicators -> same as default, but with the logic of red foreground on caps lock, green foreground on FN1 and blue foreground on FN2.
* thomazmoura -> my own profile as a sample of an ~over-engineered~ advanced case scenario.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [x] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/annepro2/keymaps/thomazmoura/rules.mk [hunk 1]
```diff
@@ -0,0 +1,47 @@
+# Anne Pro 2
```
Comment (Codetector1374) [inline keyboards/annepro2/keymaps/thomazmoura/rules.mk:1]: Is there a specific reason to include this? Having a lot of different make files could make any build change hard to maintain.

**Comment by Codetector1374** [comment]
I understand you have your profile setup as an advanced example, however if possible I think it would be nice if you can "include" the main makefile and then override some configs. (I think it should be possible)

**Comment by thomazmoura** [comment]
Sorry, I'm not sure I understood that. 

You're suggesting I make further changes to my personal profile to use it as a sample?

So far all the changes I had to make that were kind of "internal" I managed to make by adding further files to my own profile such as the thomazmoura/config.h . I've done so to be able to set them without changing the other profiles or the "core" of the QMK port. Is there another way to do it which would be more appropriate?

**Comment by thomazmoura** [comment]
Sorry, I guess I understand it now. I've referenced rules.mk to enable the tap dance feature. I'll try to enable it without the rules.mk layer (setting it on the config.h) and see if it works.

**Comment by Codetector1374** [comment]
> Sorry, I guess I understand it now. I've referenced rules.mk to enable the tap dance feature. I'll try to enable it without the rules.mk layer (setting it on the config.h) and see if it works.

That is one option. I was originally suggest that you can have a rules.mk but all you have in there is just 
```Makefile
include ../../rules.mk
FEATURE=TRUE etc...
```
so if we make changes to the SRC list etc we only have to change it at one place

**Pull request merged**